output/datadog/eventserializer.go

```diff
@@ -0,0 +1,52 @@
+package datadog
+
+import (
+	"encoding/json"
+
+	"github.com/relex/gotils/logger"
+	"github.com/relex/slog-agent/base"
+)
+
+type eventSerializer struct {
+	logger logger.Logger
+	config SerializationConfig
+	schema base.LogSchema
+}
+
+type datadogPayload struct {
```

Comment:
Datadog should be able to support free-form JSON, meaning the structure is not required. You can try to send a map created from a log record with non-empty values (empty field = skip, same as in fluentdforward) with additional DDSource and DDTags.